src/libcollections/fmt.rs

```diff
@@ -115,13 +115,8 @@
 //!
 //! If this syntax is used, then the number of characters to print precedes the
 //! actual object being formatted, and the number of characters must have the
-//! type `usize`. Although a `usize` can be printed with `{}`, it is invalid to
```

Comment:
> it is invalid to reference an argument as such.

Personally I struggled to grasp the meaning of this sentence, so I'm glad to see it'll be replaced with a slightly better alternative. But the section still contains so many problems. Perhaps all the descriptions about the variable-precision thingy should be consolidated to an independent section because it is such a unique "precision specifier" that incurs special restrictions on how to consume a "positional argument" and on the "argument type."

The "invalid format string" example surely contained some information but I'd say it would make sense only to those who already understood the internal mechanism of `std::fmt`. It was not a complete executable code snippet and required some context to make sense. But by extending the definition of the "context", I could indeed make it work: `println!("{} {:.*} {0}", 1, 2, 3.456);`